### ISSUE TITLE
VoIP UX/UI wishlist

### ISSUE THREAD

**Issue by SimonBrandner** [vector-im/element-web#16084]
### Introduction
Currently, there is a great amount of work being done on VoIP :+1: This issue aims to, summarize the missing UX/UI bits, which I think are required to reach the *perfect* user experience. I will update this as new issues are created, and I get new ideas.

**Note:** Some of the things mentioned here aren't useful at this point. They would become important if native Matrix groups calls were implemented. I am not sure if that is ever going to happen.

### Wishlist

+ [ ] Hide local video feed if the webcam is turned off (my [PR](https://github.com/matrix-org/matrix-react-sdk/pull/5529))
+ [ ] Implement screen sharing properly (not the current "easter egg" way)
+ [ ] Good UI/UX for switching between different streams
  + [ ] Ability to change the position of the local feed by dragging
  + [ ] Ability to hide the local feed
  + [ ] Focus on a specific feed (screen sharing, webcam)
  + [ ] Show multiple feeds at once (Zoom-style)
  + [ ] Show multiple feeds at once (current style but customizable - the user is looking at screen sharing and has remote webcam feed in the corner)
+ [ ] Ability to change speaker, mic and webcam during a call
+ [ ] The only difference between the voice/video call buttons should be the state (on/off) of the webcam  when the call starts) - the user should be able to turn on their webcam mid-call
+ [ ] Show call length/duration/elapsed time (Android [PR](https://github.com/vector-im/element-android/pull/2575))
+ [ ] Control volume of call participants

### Related
[Voice/Video Conference Wishlist](https://github.com/vector-im/element-web/issues/3025)

**Comment by K1aymor** [comment]
At least for me, individual volume sliders for each person in a call is really important. Some people's mics are very loud, and some people are very quiet (especially on phones). Being able to simply right-click on someone and adjust their volume is really nice in Discord, while group calls in Zoom are really annoying because some people are way louder than others and you can't adjust their volumes to compensate for it.

All the other wishlist items would be great to have, but they seem more like annoyances that people can work around, while individual volume sliders is necessary to be able to hear everyone in a group call.

**Comment by t3chguy** [comment]
> At least for me, individual volume sliders for each person in a call is really important. Some people's mics are very loud, and some people are very quiet (especially on phones). Being able to simply right-click on someone and adjust their volume is really nice in Discord, while group calls in Zoom are really annoying because some people are way louder than others and you can't adjust their volumes to compensate for it.
> 
> All the other wishlist items would be great to have, but they seem more like annoyances that people can work around, while individual volume sliders is necessary to be able to hear everyone in a group call.

Jitsi has this feature.

**Comment by SimonBrandner** [comment]
Though, we might need this when I get to audio screen-sharing

**Comment by Vaelio** [comment]
Maybe some sort of microphone filters aswell ? Volume gate, background noise removal, echo cancellation etc

**Comment by SimonBrandner** [comment]
> Maybe some sort of microphone filters aswell ? Volume gate, background noise removal, echo cancellation etc

This issue is more about the UI than the functionality itself, a noise gate has already been requested in another issue

**Comment by wchao** [comment]
I agree it would be great to have the ability to change the speaker and microphone devices mid-call.

Another interesting feature would be to hand off the call to a different Element session on another device (same user, same room). In Jitsi, Zoom, Teams, etc., it's easy to drop off and connect via a different device without disrupting the conference for other participants, but for a 1:1 call, this means hanging up the call and calling the person back.

**Comment by SimonBrandner** [comment]
@wchao, please open a new issue/look for existing ones. This issue is _only_ about the UI.

**Comment by speatzle** [comment]
Muting / Independent Volume control for screen shares would be a very good feature

**Comment by SimonBrandner** [comment]
> Muting / Independent Volume control for screen shares would be a very good feature

That is in https://github.com/vector-im/element-web/issues/16657

**Comment by Thatoo** [comment]
In a professional environment, I'd appreciate not only that any participant can share his/her own screen, but that we can allow any other participant to interact with this shared screen by drawing and writing a note on it.

**Comment by aleld** [comment]
I really love the direction element is heading regarding the video rooms. That being sad I really want to use it as my discord replacement. But the one thing that is holding me back is that you can not see who is currently joined the "channel" like actively atm. Currently there is a small icon with a number beside the roomname which shows how many people are active in that video room atm. Its something, but to be truly useful I need to see the names in order to decide whether I want to join in or not.

The moment this happens I can move my whole Discord server + friends to Element which I'd love to do.

**Comment by birbowirbo** [comment]
Hi I would love the ability to voice ONLY rooms. Functionally removing the ability for people to turn on a camera, or default them to camera off. 
This would be close to my favorite functionality of discord which is the persistent voice channels.
